src/gwt/src/org/rstudio/studio/client/workbench/addins/Addins.java

```diff
@@ -73,6 +84,47 @@ public final static RAddin decode(String decoded)
       protected RAddins() {}
    }
    
+   public static class AddinExecutor
+   {
+      public AddinExecutor()
+      {
+      }
+      
+      @Inject
+      private void initialize(AddinsServerOperations server, EventBus events)
+      {
+         server_ = server;
+         events_ = events;
+      }
+      
+      public void execute(RAddin addin)
+      {
+         if (!injected_)
```

Comment:
Sometimes you can solve these sorts of injection ordering problems using Provider<T> (which then calls for the class only when it's actually needed rather than at construction time)